[quant][graph] Add pass to insert quant dequant for dynamic quantization

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35366 [quant][graph] Add pass to insert quant dequant for dynamic quantization**
* #35325 [quant][graph] Update dynamic quant tests to use new qconfig
* #35265 [quant][graph] Add a new observer type for dynamic quantization
* #35235 [quant][graph] Add _choose_qparams function for graph mode

Summary:
Add _choose_qparams_per_tensor which returns scale and zero_point similar to the dynamic quantization in the operator

Test Plan:
python test/test_quantize_script.py
Reviewers:

Subscribers:

Tasks:

Tags:

supriyar committed:
[quant][graph] Add pass to insert quant dequant for dynamic quantization
Summary:
Add _choose_qparams_per_tensor which returns scale and zero_point similar to the dynamic quantization in the operator

Test Plan:
python test/test_quantize_script.py
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graph] Add pass to insert quant dequant for dynamic quantization"


Summary:
Add _choose_qparams_per_tensor which returns scale and zero_point similar to the dynamic quantization in the operator

Test Plan:
python test/test_quantize_script.py
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -1231,6 +1231,76 @@ void insertQuantDeQuantCall(
   insertDeQuantCall(g, quant->output(), v, uses);
 }
 
+void insertChooseQParamsQuantDeQuantCall(
```
can you refactor `insertQuantDeQuantCall` and extract the common parts instead of copy the code?

jerryzh168 replied:
actually I think you can just merge this function into `insertQuantDeQuantCall`

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -1231,6 +1231,76 @@ void insertQuantDeQuantCall(
+
+  Node *quant, *choose_qparams;
+
+  if (v->node()->kind() != prim::GetAttr){
```
I think we'll need more reliable checks for activation

supriyar committed:
Update on "[quant][graph] Add pass to insert quant dequant for dynamic quantization"


Summary:
Add _choose_qparams_per_tensor which returns scale and zero_point similar to the dynamic quantization in the operator

Test Plan:
python test/test_quantize_script.py
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graph] Add pass to insert quant dequant for dynamic quantization"


Summary:
Add _choose_qparams_per_tensor which returns scale and zero_point similar to the dynamic quantization in the operator

Test Plan:
python test/test_quantize_script.py
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
